addons/payment_adyen_paybylink/models/payment_acquirer.py

```diff
@@ -0,0 +1,144 @@
+from odoo.exceptions import ValidationError
+
+from odoo.addons.payment_adyen_paybylink.const import API_ENDPOINT_VERSIONS
+
+
+_logger = logging.getLogger(__name__)
+
+
+class PaymentAcquirer(models.Model):
+    _inherit = 'payment.acquirer'
+
+    adyen_api_key = fields.Char(
+        string='API Key', help="The API key of the webservice user", required_if_provider='adyen',
+        groups='base.group_user'
+    )
+    adyen_hmac_key = fields.Char(
+        string="HMAC Key", help="The HMAC key of the webhook", required_if_provider='adyen',
+        groups='base.group_user'
+    )
+    adyen_checkout_api_url = fields.Char(
+        string="Checkout API URL", help="The base URL for the Checkout API endpoints",
+        required_if_provider='adyen',
+    )
+    # We set a default for the now unused key fields rather than making them not required to avoid
+    # the error log at DB init when the ORM tries to set the 'NOT NULL' constraint on those fields.
+    adyen_skin_code = fields.Char(default="Do not use this field")
+    adyen_skin_hmac_key = fields.Char(default="Do not use this field")
+
+    @api.model_create_multi
+    def create(self, values_list):
+        for values in values_list:
+            self._adyen_trim_api_urls(values)
+        return super().create(values_list)
+
+    def write(self, values):
+        self._adyen_trim_api_urls(values)
+        # We set a default for the now unused key fields rather than making them not required to avoid
+        # the error log at DB init when the ORM tries to set the 'NOT NULL' constraint on those fields.
+        values.update(
+            adyen_skin_code="Do not use this field",
+            adyen_skin_hmac_key="Do not use this field",
+        )
+        return super().write(values)
+
+    @api.model
+    def _adyen_trim_api_urls(self, values):
+        """ Remove the version and the endpoint from the url of Adyen API fields.
+        :param dict values: The create or write values
+        :return: None
+        """
+        for field_name in ('adyen_checkout_api_url'):
+            if values.get(field_name):  # Test the value in case we're duplicating an acquirer
+                values[field_name] = re.sub(r'[vV]\d+(/.*)?', '', values[field_name])
+
+    def adyen_form_generate_values(self, values):
+        base_url = self.get_base_url()
+
+        paymentAmount = self._adyen_convert_amount(values['amount'], values['currency'])
+        adyen_paybylink_data = {
+            'reference': values['reference'],
+            'amount': {
+                'value': '%d' % paymentAmount,
+                'currency': values['currency'] and values['currency'].name or '',
+            },
+            'merchantAccount': self.adyen_merchant_account,
+            'shopperLocale': values.get('partner_lang', ''),
+            'returnUrl': urls.url_join(base_url, '/payment/process'),
+            'shopperEmail': values.get('partner_email') or values.get('billing_partner_email') or '',
```

Comment:
Well I'll pass on this one ':)

Reply:
Not even for:
```suggestion
            'shopperEmail': values.get('partner_email') or values.get('billing_partner_email', ''),
```
(Well, if Adyen doesn't send `{'billing_partner_email' : none}` or something like that when there isn't one at least, because then my version wouldn't work :grin: )

Reply:
Not even for:
```suggestion
            'shopperEmail': values.get('partner_email') or values.get('billing_partner_email', ''),
```
(Well, if Adyen doesn't send `{'billing_partner_email' : none}` or something like that when there isn't one at least, because then my version wouldn't work :grin: )
Anyway, your version is clearer and safer :laughing: